docs/general/ad-filtering/create-own-filters.md

```diff
@@ -865,380 +937,466 @@ For the requests matching a `$csp` rule, we will strengthen response security po
+    ]
+}
 ```
 
-> **Compatibility with different versions of AdGuard**
->
-> Rules with `$denyallow` modifier are not supported by AdGuard Content Blocker.
-
-#### **`$removeparam`** {#removeparam-modifier}
+</details>
 
-> `$queryprune` is an alias of `$removeparam`. Since `$queryprune` is deprecated, avoid using it and use `$removeparam` instead.
+#### **`$network`** {#network-modifier}
 
-Rules with `$removeparam` modifier are intended to strip query parameters from requests' URLs. Please note that such rules are only applied to `GET`, `HEAD`, `OPTIONS`, and [sometimes](#removeparam-modifier-compatibility) `POST` requests.
+This is basically a Firewall-kind of rules allowing to fully block or unblock access to a specified remote address.
 
-> `$removeparam` rules that do not have any [content type modifiers](#content-type-modifiers) will match only requests where content type is `document`.
+1. `$network` rules match **IP addresses only**! You cannot use them to block or unblock access to a domain.
+2. To match an IPv6 address, you have to use the collapsed syntax, e.g. use `[2001:4860:4860::8888]$network` instead of `[2001:4860:4860:0:0:0:0:8888]$network`.
+3. An allowlist `$network` rule makes AdGuard bypass data to the matching endpoint, e.g. there will be no further filtering at all.
+4. If the IP part starts and ends with `/` character, it is treated as a regular expression.
 
-**Syntax**
+> We recommend to get acquainted with this [article](#regexp-support), for better understanding of regular expressions.
 
-**Basic syntax**
+**Examples**
 
-* `$removeparam=param` — removes query parameter with the name `param` from URLs of any request, e.g. a request to `http://example.com/page?param=1&another=2` will be transformed to `http://example.com/page?another=2`.
+* `174.129.166.49:3478^$network` blocks access to `174.129.166.49:3478` (but not to `174.129.166.49:34788`).
+* `[2001:4860:4860::8888]:443^$network` blocks access to `[2001:4860:4860::8888]:443`.
+* `174.129.166.49$network` blocks access to `174.129.166.49:*`.
+* `@@174.129.166.49$network` makes AdGuard bypass data to the endpoint. No other rules will be applied.
+* `/.+:3[0-9]{4}/$network` blocks access to any port from 30000 to 39999.
+* `/8.8.8.(:?8|4)/$network` blocks access to both `8.8.8.8` and `8.8.8.4`.
 
-> `$removeparam` basic syntax is supported starting with v1.7 of [CoreLibs](https://adguard.com/en/blog/introducing-corelibs.html) and v3.6 of AdGuard Browser Extension.
+> **Compatibility with different versions of AdGuard**
+>
+> Only AdGuard for Windows, Mac, Android are technically capable of using rules with `$network` modifier.
 
-**Regular expressions**
+#### **`$permissions`** {#permissions-modifier}
 
-You can also use regular expressions to match query parameters and/or their values:
+This modifier completely changes the rule behavior. If it is applied to a rule, it will not block the matching request. The response headers are going to be modified instead.
```

Comment:
> If it is applied to a rule, it will not block the matching request.
имеется в виду, что правило не будет блокировать запрос? тогда лучше If it is applied to a rule, the rule will not block the matching request. Response headers will be modified instead.